src/api/bulk.ts

```diff
@@ -876,6 +1007,657 @@ export class Bulk<S extends Schema> {
   }
 }
 
+export class BulkV2<S extends Schema> {
+  #connection: Connection<S>;
+
+  /**
+   * Polling interval in milliseconds
+   */
+  pollInterval = 1000;
+
+  /**
+   * Polling timeout in milliseconds
+   * @type {Number}
+   */
+  pollTimeout = 10000;
+
+  constructor(connection: Connection<S>) {
+    this.#connection = connection;
+  }
+
+  /**
+   * Create a new job instance
+   */
+  createJob<Opr extends IngestOperation>(
+    options: NewIngestJobOptions,
+  ): IngestJobV2<S, Opr> {
+    return new IngestJobV2({
+      connection: this.#connection,
+      jobInfo: options,
+      pollingOptions: this,
+    });
+  }
+
+  job<Opr extends IngestOperation>(
+    options: ExistingIngestJobOptions,
+  ): IngestJobV2<S, Opr> {
+    return new IngestJobV2({
+      connection: this.#connection,
+      jobInfo: options,
+      pollingOptions: this,
+    });
+  }
+
+  /**
+   * Create, upload, and start bulkload job
+   */
+  async load(
```

Comment:
Why `load` instead of `upload`? Wouldn't that more accurately describe what this method does?

Reply:
`load` was chosen to maintain consistency with the V1 implementation (see https://jsforce.github.io/jsforce/doc/Bulk.html#load).  i'm open to alternative name suggestions 😄